### PR TITLE
[FLINK-20520][metrics][docs] Document metric name escaping

### PR DESCRIPTION
**Pull request by zentol**

### CODE REVIEW

File: docs/ops/metrics.md [hunk 1]
```diff
@@ -1654,6 +1654,9 @@ Request metrics aggregated over a subset of all entities of the respective type:
   - `/jobs/metrics?jobs=D,E,F`
   - `/jobs/<jobid>/vertices/<vertexid>/subtasks/metrics?subtask=1,2,3`
 
+<span class="label label-danger">Warning</span> Metric names can contain special characters (`?`, `&`, `$`, `+`, `/`, `,`, `:`, `=`, `@`, `%`) that you need to be escape when querying metrics.
+For example, `"a_+_b`" would be escaped to "`a_%2B_b`".
```
Comment (V1ncentzzZ) [inline docs/ops/metrics.md:1658]: Maybe  "\`a_+\_b\`" instead of \`"a_+_b\`"?

File: docs/ops/metrics.zh.md [hunk 2]
```diff
@@ -1654,6 +1654,9 @@ Request metrics aggregated over a subset of all entities of the respective type:
   - `/jobs/metrics?jobs=D,E,F`
   - `/jobs/<jobid>/vertices/<vertexid>/subtasks/metrics?subtask=1,2,3`
 
+<span class="label label-danger">Warning</span> Metric names can contain special characters (`?`, `&`, `$`, `+`, `/`, `,`, `:`, `=`, `@`, `%`) that you need to be escape when querying metrics.
+For example, `"a_+_b`" would be escaped to "`a_%2B_b`".
```
Comment (V1ncentzzZ) [inline docs/ops/metrics.zh.md:1658]: Same problem as before.

File: docs/ops/metrics.md [hunk 3]
```diff
@@ -1654,6 +1654,9 @@ Request metrics aggregated over a subset of all entities of the respective type:
   - `/jobs/metrics?jobs=D,E,F`
   - `/jobs/<jobid>/vertices/<vertexid>/subtasks/metrics?subtask=1,2,3`
 
+<span class="label label-danger">Warning</span> Metric names can contain special characters (`?`, `&`, `$`, `+`, `/`, `,`, `:`, `=`, `@`, `%`) that you need to be escape when querying metrics.
```
Comment (tillrohrmann) [inline docs/ops/metrics.md:1657]:
```suggestion
<span class="label label-danger">Warning</span> Metric names can contain special characters (`?`, `&`, `$`, `+`, `/`, `,`, `:`, `=`, `@`, `%`) that you need to escape when querying metrics.
```

Comment (tillrohrmann) [reply docs/ops/metrics.md:1657]: Can we link to the rules how to escape these characters, e.g. a table of URL Escape Characters?

**Review by tillrohrmann — COMMENTED** [review]
Thanks for creating this PR @zentol. The change itself looks good. I had two comments. Please take a look.

File: docs/ops/metrics.md (hunk 3 above)
Comment (zentol) [reply docs/ops/metrics.md:1657]: I had hoped that the example gives users enough of a reference to find a table.

The only _official_(?) reference I could find for which characters to escape is this [RFC](https://tools.ietf.org/html/rfc3986#section-2.2), but it's not as readable as I would like.
[This](https://secure.n-able.com/webhelp/NC_9-1-0_SO_en/Content/SA_docs/API_Level_Integration/API_Integration_URLEncoding.html) page was the basis for which characters I explicitly listed, but it is not really consistent with the RFC.
I originally did not want to list any characters at all, but with the RFC being _eh_ and all other sources not being _official_ I couldn't find solution I really liked.

As for how to escape, it is essentially just "% + ascii_hex(character)", which I can add.

Comment (tillrohrmann) [reply docs/ops/metrics.md:1657]: Can we then maybe create the table for the characters you have listed?

**Review by tillrohrmann — APPROVED** [review]
Thanks for updating this PR @zentol. LGTM. +1 for merging.

**Pull request merged**